jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -1151,11 +1151,24 @@ public static void loadLibrary(String libName) {
 	if (smngr != null) {
 		smngr.checkLink(libName);
 	}
+	try {
+/*[IF JAVA_SPEC_VERSION >= 15]*/
+		ClassLoader.loadLibrary(getCallerClass(), libName);
```

Comment:
Pls cache the result of getCallerClass() / callerClassLoader() since these are expensive calls.

Reply:
Cached the results.